### ISSUE TITLE
Create a new state to WorkQueue Element called "New"

### ISSUE THREAD

**Issue by ticoann** [dmwm/WMCore#8135]
Let me repurpose this issue as the following:

We need to create a new WorkQueue Element status named `New`, as the states are defined here:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/DataStructs/WorkQueueElement.py#L9

and this status will be the first status for any new elements created in the **global workqueue**.
When elements are created in the agent (sigh..., yes, we have this over-complication!), they can be directly created in the Available status. 

In addition to the extra status, we should create a new WQE attribute to keep track of the subscription id.

Original issue was (keeping here only for the record):
"""
1. When request is assignment-approved status GQ needs to be populated with elements in Hold state.
2. The elements in "Hold" states shouldn't be replicated to local WQ.
3. When request is change to "assigment" state, it should change GQE state to "Availabe" state.

Need to have strict condition for the requests to move to assignment-approved state. Otherwise there might be too many GQ elements created. (might need intermediate status between new to assignment-approved state0 
"""

**Comment by amaltaro** [comment]
We should probably create a wiki explaining how exactly it's supposed to work, before any work gets done. As you said, it can easily become a bottleneck.

**Comment by ticoann** [comment]
Hi Alan, I started here.  feel free to modify.
https://github.com/dmwm/WMCore/wiki/Unified-Porting

**Comment by amaltaro** [comment]
Ok, let me get this one started then. Maybe beginning of the next week once integration phase is over.

**Comment by amaltaro** [comment]
I see two possible use cases for this issue:
1. only Hold workflows that haven't been assigned yet (pointing to a possible issue with the request)
2. also Hold workflows after they were assigned, so trying to hold work for that workflow already in the system.

Case 1. should be simple, not sure how useful it's though.
Case 2. is more challenging and requires further discussion. Is it enough to stop WQE replication? Or we should also stop job creation and job submission?

**Comment by amaltaro** [comment]
(changed my mind a few times on this topic, here is how I think we should proceed)

1. Unified assigns a workflow (request transition from "assignment-approved" to "assigned")
2. global workqueue picks up requests in "assigned" status, and does its usual business creating all the GQEs, however, it creates elements in the "New" status. Once it's completed, the request is moved from "assigned" to "acquired"
3. then the microservice kicks in and query ReqMgr2 for requests in "acquired" status; given the list of requests in that status, microservice queries for their GQEs in the New status:
3.a. set the GQE to "Available" if it doesn't have any input AND secondary data
3.b. according to the SiteWhitelist and data location (input + parents + pileup), check the campaign configuration stored in Couch, run the algorithm ported from Unified and queue blocks and the location they will get subscribed
3.c. once we have the block x location map for all GQEs within a given request, the concurrent PhEDEx subscription is executed
3.d. once all subscriptions are successful, update the GQE status from "New" to "Available"
3.e. last but not least, update the request from "acquired" to "staging" status
3.f. repeat the same for all workflows
4. at this point, those GQE are eligible to be picked up by any agent.

If there is a problem creating the subscriptions, we can either:
a) retry it once
b) bail out (no status transitions at request neither at GQ level) and retry in the next daemon cycle

I'll document it later and also repurpose this ticket.

**Comment by vkuznet** [comment]
Fine with me, but I don't see much changes needed for ReqMgr2ms since in high level view it can perform these actions, i.e. talk to ReqMgr2 and request something for specific status, query GQE, get subscriptions. What you're showing is concrete plan for specific actions, but architecture wise the MS is capable of doing that.

**Comment by amaltaro** [comment]
We have discussed a new architecture for the MS Transferor module and this new GQE status is no longer needed. closing
